File: tests/fixtures/csharp/Sample.cs
using System;
using System.Collections.Generic;

namespace Example
{
    public interface IGreetable
    {
        string Greet();
    }

    public enum Status
    {
        Active,
        Inactive
    }

    public class UserService : BaseService, IGreetable
    {
        private string _name;

        public UserService(string name)
        {
            _name = name;
        }

        public string GetName()
        {
            return _name;
        }

        public string Greet()
        {
            return "Hello, " + GetName();
        }

        public string Classify(int x)
        {
            if (x > 0)
            {
                return "positive";
            }
            else if (x < 0)
            {
                return "negative";
            }
            return "zero";
        }
    }

    public class UserServiceTests
    {
        [Fact]
        public void TestGetName()
        {
            var svc = new UserService("test");
            Assert.Equal("test", svc.GetName());
        }
    }
}

File: tests/fixtures/rust/sample.rs
use std::fmt;
use crate::utils::helper;

pub trait Greetable {
    fn greet(&self) -> String;
}

pub struct UserService {
    name: String,
}

pub enum Status {
    Active,
    Inactive,
}

impl UserService {
    pub fn new(name: String) -> Self {
        UserService { name }
    }

    pub fn get_name(&self) -> &str {
        &self.name
    }
}

impl fmt::Display for UserService {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{}", self.name)
    }
}

#[test]
fn test_new_user_service() {
    let svc = UserService::new("test".to_string());
    assert_eq!(svc.get_name(), "test");
}

pub fn classify(x: i32) -> &'static str {
    if x > 0 {
        "positive"
    } else if x < 0 {
        "negative"
    } else {
        "zero"
    }
}
